### ISSUE TITLE
pod lint doesn't respect swift_version of dependencies

### ISSUE THREAD

**Issue by sprynmr** [CocoaPods/CocoaPods#8940]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Ran `pod lib lint mypod`

## What did you expect to happen?

Dependencies with different versions of swift to compile with their respective versions and pass linting.

Per discussion here: https://github.com/CocoaPods/CocoaPods/pull/8897#issuecomment-506399057

## What happened instead?

Linting failed as it was using the swift version from the main Podspec to compile the dependencies

### Stack

```
   CocoaPods : 1.7.2
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.14.5 (18F203)
       Xcode : 10.2.1 (10E1001)
         Git : git version 2.20.1 (Apple Git-117)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ a551a2a5f25a47ddf704f7937f8c617dd119f56b
          
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
claide-plugins                   : 0.9.2
cocoapods-deintegrate            : 1.0.4
cocoapods-plugins                : 1.0.0
cocoapods-search                 : 1.0.0
cocoapods-static-swift-framework : 0.4
cocoapods-stats                  : 1.1.0
cocoapods-trunk                  : 1.3.1
cocoapods-try                    : 1.1.0
```

## Project that demonstrates the issue

https://twstaticfiles.s3.amazonaws.com/specbug.zip

**Comment by dnkoutso** [comment]
@sprynmr that is a confirmed bug. Will ship in 1.7.3.

The PR #8897 actually regressed it. Thanks again for filing.

**Comment by dnkoutso** [comment]
@sprynmr have you tried CocoaPods 1.7.1?

**Comment by sprynmr** [comment]
I haven't, but I will test it now.

**Comment by dnkoutso** [comment]
Fix https://github.com/CocoaPods/CocoaPods/pull/8941

**Comment by dnkoutso** [comment]
![Screen_Shot_2019-06-27_at_1_23_33_PM](https://user-images.githubusercontent.com/310370/60286990-c82c0300-98de-11e9-8b34-d9a1a40d3767.png)

Confirmed in my change.

**Comment by sprynmr** [comment]
🎉

**Comment by sprynmr** [comment]
Confirmed 1.7.1 also does not have the issue.

**Comment by dnkoutso** [comment]
Cool.

**Comment by dnkoutso** [comment]
Will ship with 1.7.3 which also has some improvements for CDN in CocoaPods/Core.

**Comment by dnkoutso** [comment]
Shipped 1.7.3.
